src/data/nat/fib.lean

```diff
@@ -96,4 +97,70 @@ begin
     mono }
 end
 
-end nat
+/-- Subsequent Fibonacci numbers are coprime,
+  see https://proofwiki.org/wiki/Consecutive_Fibonacci_Numbers_are_Coprime -/
+lemma fib_coprime_fib_succ (n : ℕ) : nat.coprime (fib n) (fib (n + 1)) :=
+begin
+  unfold coprime,
+  induction n with n ih,
+  { simp },
+  { convert ih using 1,
+    rw [fib_succ_succ, succ_eq_add_one, gcd_rec, add_mod_right, gcd_comm (fib n),
+      gcd_rec (fib (n + 1))], }
+end
+
+/-- See https://proofwiki.org/wiki/Fibonacci_Number_in_terms_of_Smaller_Fibonacci_Numbers -/
+lemma fib_add (m n : ℕ) :
+  fib m * fib n + fib (m + 1) * fib (n + 1) = fib (m + n + 1) :=
+begin
+  induction n with n ih generalizing m,
+  { simp },
+  { intros,
+    specialize ih (m + 1),
+    rw [add_assoc m 1 n, add_comm 1 n] at ih,
+    simp only [fib_succ_succ, ← ih],
+    ring, }
+end
+
+
+lemma gcd_fib_add_self' (m n : ℕ) : gcd (fib m) (fib (n + m)) = gcd (fib m) (fib n) :=
+begin
+  cases eq_zero_or_pos n,
+  { rw h, simp },
+  replace h := nat.succ_pred_eq_of_pos h, rw [← h, succ_eq_add_one], symmetry,
+  calc (fib m).gcd (fib (n.pred + 1)) 
+        = gcd (fib m) (fib (n.pred + 1) * fib (m + 1)) :
+    begin
+      have hcop := coprime.symm (fib_coprime_fib_succ m),
+      rwa [gcd_comm, gcd_comm (fib m) _, mul_comm _ (fib (m + 1)), coprime.gcd_mul_left_cancel]
+    end
+    ... = gcd (fib m) (fib (n.pred) * (fib m) + fib (n.pred + 1) * fib (m + 1)) :
+    by rw [← gcd_add_self (fib m) _ (fib (n.pred)), add_comm]
+    ... = m.fib.gcd (n.pred + 1 + m).fib :
+    by { rw (fib_add n.pred _), ring }
+    ... = m.fib.gcd (n.pred + 1 + m).fib : by ring,
```

Comment:
This step seems to be identical to the one above it. So I guess you can shorten this `calc`-block by two lines.